akka-actor-tests/src/test/scala/akka/pattern/AskSpec.scala

```diff
@@ -256,6 +256,29 @@ class AskSpec extends AkkaSpec("""
       val (promiseActorRefForSelection, _) = p.expectMsgType[(ActorRef, String)]
       promiseActorRefForSelection.path.name should startWith("_user_myName")
     }
+
+    "proper path when promise actor terminated" in {
+      implicit val timeout: Timeout = Timeout(300 millis)
+      val p = TestProbe()
+
+      val act = system.actorOf(Props(new Actor {
+        def receive = {
+          case _ =>
+            val senderRef: ActorRef = sender()
+            senderRef ! "complete"
+            p.ref ! senderRef
+        }
+      }), "pathPrefix")
+
+      val f = (act ? "ask").mapTo[String]
+      val promiseActorRef = p.expectMsgType[ActorRef]
+
+      // verify ask complete
+      val completed = f.futureValue
+      completed should ===("complete")
+      // verify path was proper
+      promiseActorRef.path.toString should include("pathPrefix")
+    }
```

Comment:
Fix looks good but I don't see how this tests asking a stopped actor?

Reply:
Ah, sorry, tricky test, we are returning sender to the outside, and the original ask already stopped. This is good.